### ISSUE TITLE
Support BeerCoin

### ISSUE THREAD

**Issue by friedger** [walleth/walleth#193]
For BeerCoin details see https://www.reddit.com/r/ethereum/comments/4v7opj/introducing_beercoin/

Currently, it is not possible to transfer a received BeerCoin to someone else. It would be nice as well to see who owes you a BeerCoin and who you owe a BeerCoin. This could be an example for a plugin mechanism for contracts that require some special UI.

**Comment by ligi** [comment]
thanks for the sugestuin

> Currently, it is not possible to transfer a BeerCoin if the user has not yet received one as the UI requires a positive balance of tokens for that.

showing the create transfer icon should be determined by the ETH balance not the token balance. You need eth to pay the gas (In all scenarios I can currently come up with - including BeerCoin)

> Furthermore, it is not possible to transfer a received BeerCoin to someone else.

can you elaborate why? should be a normal ERC-20 transfer or am I missing something?

But that might not be wanted here this way - I think this goes into fungible tokens (https://github.com/ethereum/EIPs/issues/721) - a beer coin by vitalik buterin might have different value than the one from donald dumpf ..-)

**Comment by friedger** [comment]
I don't know whether BeerCoin fits into any classification. 

As I understand, the rules are:
- User A can send a BeerCoin to anybody, but just one (the default max count).
- User B can send the BeerCoin back to user A as soon as user A bought user B a beer.
- User B can send the BeerCoin, received from user A, to user C. (This requires a third address in the token transfer and is a custom transaction IMO)

I remove the first bit of the issue into a new issue.

**Comment by ligi** [comment]
> This requires a third address in the token transfer and is a custom transaction IMO

why do you think a custom transaction? should be a normal ERC-20 transfer()

my main question is how to differentiate between different BeerCoins - as  @arachnid states correctly: 

> A person's BeerCoin is only as valuable as the recipient's belief that they're good for the beer, should it ever be redeemed; a beer owed to you by Vitalik Buterin is probably worth more than a beer owed to you by the DAO hacker (but your opinions may differ on that point).

So if you transfer() to C and C transfer()s to D - D needs a way to know if the source was A or B to know who to go to for a :beer: 

Anyway the BeerCoin coin is already listed:

https://github.com/ethereum-lists/tokens/blob/72d5f1547e6281fb1c1e21238651477c2957d747/tokens/eth/0x74C1E4b8caE59269ec1D85D3D4F324396048F4ac.json

 but would be great to have it on rinkeby - only see ABI but no source currently in the thread.

**Comment by friedger** [comment]
Here is the source code: 
https://etherscan.io/address/0x74c1e4b8cae59269ec1d85d3d4f324396048f4ac#code

D can see the source by querying the contract. The tansfer of debts keeps the information about the source.

**Comment by ligi** [comment]
ah right - yea but I think we should really got the ERC-721 with ERC-165 route here to not have these special cases in code. They have not been available at the time this contract was written

**Comment by Arachnid** [comment]
These aren't NFTs, they're a mutual currency. I don't think anyone's written a token standard for that yet, unfortunately.

I agree that adding support for just one token without a standard is probably a waste of time, though.

**Comment by friedger** [comment]
@Arachnid Thanks for clarification.

I close this issue (BeerCoin should be usable after #195 is fixed) and suggest to first add support for interfaces. Then we can see what interfaces become popular (independent of whether they implement ERC-165 or not) and provide an appropriate UI. 

The information of supported interfaces should be stored in the token list so that the discovery is not done again and again.
